crates/primitives/src/transaction/mod.rs

```diff
@@ -57,10 +59,12 @@ pub mod signature;
 pub(crate) mod util;
 mod variant;
 
-#[cfg(feature = "optimism")]
+#[cfg(all(feature = "optimism", not(feature = "scroll")))]
```

Comment:
Do we need to add this everywhere to keep tests passing? Can't we require that at most one of these are enabled?

Reply:
I was thinking maybe you could just use two variant with the same value behind different feature flags, and make sure that those two features flags [are never both enabled](https://stackoverflow.com/a/65863419) (but maybe this is not possible).

Reply:
I think it would work for everything except the clippy check (`cargo +nightly clippy --workspace --lib --examples --tests --benches --all-features --locked`), which compiles with `all-features`.

Reply:
maybe we can config the default feature in `Cargo.toml`

Reply:
I found that `cargo build` and `cargo clippy` will use the default features in `Cargo.toml`. like this

```
[features]
default = ["scroll"] 
scroll = []
```

Reply:
Oh right but I think because the CI runs `clippy` with the `--all-features` flag, if we have for example:
```
[features]
default = ["scroll"] 
scroll = []
optimism = []
```
the `optimism` feature will also be activated, even if not in the `default`, because of the `--all-features`.

Reply:
Yes, maybe we can diable the `--all-features`. I think we can put features in the `cargo.toml` instead of in the command-line args, this can keep the clippy, test, and build with the same config option.